# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

***HYPOTHESES:***

***Null hypothesis (H0)***: there is no difference in average weight loss between groups(diets).

***Alternative hypothesis (H1)***: there is a difference in average weight loss between both groups (diets). 

**Student's t-test**

In [9]:
import scipy.stats as stats

diet_1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet_2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]

# Student's t-test 
t_stat, p_value = stats.ttest_ind(diet_1, diet_2)
print(f"t_stat = {t_stat}")
print(f"p_value = {p_value}")

t_stat = -3.5383407969933938
p_value = 0.004083270191713912


***Conclusions -*** since we got a p_value (0.004) lower that the chosen significance level of 5% (0.05), the null hypothesis will be rejected, concluding that there is a difference in the average weight loss between both groups (diets).

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

***HYPOTHESES:***

***Null hypothesis (H0)***: there is no difference in average corn yield  between the three types of fertilizers.

***Alternative hypothesis (H1)***: at least one of the fertilizers is superior in terms of corn production.

# ANOVA test

In [7]:
from scipy.stats import f_oneway

fertilizer_1 = [20, 21, 20, 19, 20]
fertilizer_2 = [22, 21, 23, 22, 21]
fertilizer_3 = [24, 23, 22, 23, 24]

# Anova test
f_stat, p_value = f_oneway(fertilizer_1, fertilizer_2, fertilizer_3)

print(f"f_stat: {f_stat}")
print(f"p_value: {p_value}")

f_stat: 20.315789473684188
p_value: 0.00014047824793190475


***STATEMENTS -***
The f statistic and p-value turn out to be equal to 20.32 and 0.00014, respectivelly. Since the p-value is less than 0.05 hence we would reject the null hypothesis. This implies that we have suficient proof to say that there exist a difference in the average corn yield among the three different fertilizers.


In [14]:
import numpy as np
import pandas as pd
from scipy import stats
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import matplotlib.pyplot as plt

fertilizer_1 = [20, 21, 20, 19, 20]
fertilizer_2 = [22, 21, 23, 22, 21]
fertilizer_3 = [24, 23, 22, 23, 24]

data = pd.DataFrame({
    "Valor" : np.concatenate([fertilizer_1, fertilizer_2, fertilizer_3]),
    "Grupo" : np.repeat(['Fertilizer 1', 'Fertilizer 2', 'Fertilizer 3'], repeats=5)
})
tukey = pairwise_tukeyhsd(endog=data['Valor'], groups=data['Grupo'], alpha=0.05)
print(tukey)

     Multiple Comparison of Means - Tukey HSD, FWER=0.05      
   group1       group2    meandiff p-adj  lower  upper  reject
--------------------------------------------------------------
Fertilizer 1 Fertilizer 2      1.8 0.0099 0.4572 3.1428   True
Fertilizer 1 Fertilizer 3      3.2 0.0001 1.8572 4.5428   True
Fertilizer 2 Fertilizer 3      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------------------


***CONCLUSIONS -*** From the results obtained, we can see that the three fertilizers differ significantly from each other. Fertilizer 3 is the best fertilizer in terms of average corn yield (in Kg), followed by Fertilizer 2, and the worst is Fertilizer 3.